File: server/src/main/java/com/kruna1pate1/pictionaryserver/dto/GameDto.java
package com.kruna1pate1.pictionaryserver.dto;

import com.kruna1pate1.pictionaryserver.model.enums.ServerCode;

/**
 * Created by KRUNAL on 30-06-2022
 */
public record GameDto<T>(
        ServerCode code,
        T data
) {
}

File: client/mobile/app/src/main/java/com/kruna1pate1/pictionaryapp/data/dto/GameDto.kt
package com.kruna1pate1.pictionaryapp.data.dto

import com.kruna1pate1.pictionaryapp.domain.model.Game
import com.kruna1pate1.pictionaryapp.domain.model.enums.ServerCode

data class GameDto(
    val code: ServerCode,
    val data: Any
) {
    fun <T> toGame(): Game<T> {
        return Game(
            code = code,
            data = data as T
        )
    }
}
